src/MountedTraversal.js

```diff
@@ -166,15 +194,27 @@ export function buildInstPredicate(selector) {
     if (isCompoundSelector.test(selector)) {
       return AND(splitSelector(selector).map(buildInstPredicate));
     }
-    if (selector[0] === '.') {
-      // selector is a class name
+
+    switch (selectorType(selector)) {
+    case SELECTOR.CLASS_TYPE:
       return inst => instHasClassName(inst, selector.substr(1));
-    } else if (selector[0] === '#') {
-      // selector is an id name
+    case SELECTOR.ID_TYPE:
       return inst => instHasId(inst, selector.substr(1));
+    case SELECTOR.ATTR_TYPE:
+      const attrKey = selector.split(/\[([a-z\-\:]*?)(=|\])/)[1];
```

Comment:
Good catch. That is actually a bug now that I think about it. svg introduced a lot of camelCase attributes, like `preserveAspectRatio`